hr_expense_advance_clearing/models/hr_expense.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2019 Kitti Upariphutthiphong <kittiu@ecosoft.co.th>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class HrExpense(models.Model):
+    _inherit = 'hr.expense'
+
+    advance = fields.Boolean(
+        string='Employee Advance',
+        default=False,
+    )
+
+    @api.multi
+    @api.constrains('advance')
+    def _check_advance(self):
+        emp_advance = self.env.ref('hr_expense_advance_clearing.'
+                                   'product_emp_advance')
+        if not emp_advance.property_account_expense_id:
```

Comment:
Hello @kittiu 

If you do not check the advance field, the raise not must be shown
<img width="1367" alt="Screen Shot 2019-05-04 at 11 08 18 AM" src="https://user-images.githubusercontent.com/7606656/57181835-598e7400-6e5e-11e9-9eb1-08e2766cb12e.png">
:

Reply:
Ok, I will move the checking back inside the for loop. I also move back the emp_advance back inside the for loop as well, because I think in most case, advance is not ticked, and emp_advance should not be instantiated.